bluetooth.cpp

```diff
@@ -203,16 +204,15 @@ void l2cap_accept(int ctl, int csk, int isk, int debug, int legacy)
         return;
     }
 
-#ifdef GASIA_GAMEPAD_HACKS
-    req.vendor  = 0x054c;
-    req.product = 0x0268;
-    req.version = 0x0100;
-    req.parser  = 0x0100;
+    if (get_sdp_device_info(&addr_src, &addr_dst, &req) < 0) {
```

Comment:
I'm uncertain if this function will actually fail on a Gasia controller. I know that third-party Shanwan controllers fail to pair on BlueZ due to a missing SDP record (which is patched in newer releases: https://git.kernel.org/pub/scm/bluetooth/bluez.git/commit/plugins/sixaxis.c?id=1629c39ededef07988a5403b27331e0e317f1e08).

Additionally, this may cause Shanwan controllers to use the Gasia configuration, which may be unwanted. If at least req.name is enumerated by get_sdp_device_info by a Shanwan controller, and doesn't fake the official "Sony Computer Entertainment Wireless Controller" string, then we could do a string compare against "SHANWAN" etc. to see if we really want to set gasia = 1 here.